core/include/scipp/core/dataset.h

```diff
@@ -616,7 +619,7 @@ class SCIPP_CORE_EXPORT DatasetConstProxy {
         indices.end());
     // The dimension extent is either given by the coordinate, or by data, which
     // can be 1 shorter in case of a bin-edge coordinate.
-    scipp::index extent = coords()[slice1.dim].dims()[slice1.dim];
+    scipp::index extent = currentDims.at(slice1.dim);
```

Comment:
Is this fixing #430?

Reply:
Yes. I added a new unit test that demonstrates this.